### PR TITLE
feat: create Mark and UserProfile entites, DAOs, DTOs and mappers

### PR DESCRIPTION
**Pull request by AntonKolbaska**

### CODE REVIEW

File: src/main/java/com/tproject/entity/Mark.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+@Getter
+@Setter
+@ToString
+@AllArgsConstructor
+@NoArgsConstructor
+public class Mark {
+    private int id;
+    private int userProfileId;
+    private float score;
+    private Date date;
```
Comment (yanaozembl) [inline src/main/java/com/tproject/entity/Mark.java:16]: change to LocalDate

File: src/main/java/com/tproject/dto/MarkDto.java [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+@Getter
+@Setter
+@ToString
+@AllArgsConstructor
+@NoArgsConstructor
+public class MarkDto {
+    private int id;
+    private int userProfileId;
+    private float score;
+    private Date date;
```
Comment (yanaozembl) [inline src/main/java/com/tproject/dto/MarkDto.java:16]: change to LocalDate

File: src/main/java/com/tproject/dao/impl/MarkDaoImpl.java [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+
+    private static final Logger LOGGER =
+            Logger.getLogger(MarkDaoImpl.class.getName());
+
+    private Mark composeMark(ResultSet resultSet) throws SQLException {
+        Mark mark = new Mark();
+        mark.setId(resultSet.getInt("id"));
+        mark.setUserProfileId(resultSet.getInt("user_id"));
+        mark.setScore(resultSet.getFloat("score"));
+        mark.setDate(resultSet.getDate("date"));
```
Comment (yanaozembl) [inline src/main/java/com/tproject/dao/impl/MarkDaoImpl.java:37]: is there a getLocalDate methos in resitlSet?

File: src/main/java/com/tproject/dao/impl/MarkDaoImpl.java [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+
+    @Override
+    public Optional<Mark> findMarkByUserDate(int userId, java.util.Date date) throws SQLException {
+        Optional<Mark> markOpt = Optional.empty();
+        String sql = "SELECT * FROM marks WHERE user_id = ? AND date = ?";
+
+        try (Connection conn = JdbcConnection.getInstance().getConnection();
+             PreparedStatement statement = conn.prepareStatement(sql)) {
+            statement.setInt(1, userId);
+            statement.setDate(2, new java.sql.Date(date.getTime()));    //check
```
Comment (yanaozembl) [inline src/main/java/com/tproject/dao/impl/MarkDaoImpl.java:73]: change to statement.setDate(2, date);

File: src/main/java/com/tproject/dao/impl/MarkDaoImpl.java [hunk 5]
```diff
@@ -0,0 +1,208 @@
...
+            }
+        } catch (SQLException ex) {
+            LOGGER.log(Level.SEVERE, null, ex);
+            throw new CustomSQLException("findMark - SQL Exception");
+        }
+        return markOpt;
+    }
+
+    @Override
+    public Optional<Mark> findMarkByUserDate(int userId, java.util.Date date) throws SQLException {
```
Comment (yanaozembl) [inline src/main/java/com/tproject/dao/impl/MarkDaoImpl.java:66]: change to LocalDate date

File: src/main/java/com/tproject/dao/impl/MarkDaoImpl.java [hunk 6]
```diff
@@ -0,0 +1,208 @@
...
+            throw new CustomSQLException("getAll - SQL Exception");
+        }
+
+        return marks;
+    }
+
+    @Override
+    public Optional<Integer> saveMark(Mark mark) {
+        String message = "The User to be added should not be null";
+        Mark nonNullMark = Objects.requireNonNull(mark, message);
```
Comment (yanaozembl) [inline src/main/java/com/tproject/dao/impl/MarkDaoImpl.java:118]: why do you use Objects.requireNonNull method?

File: src/main/java/com/tproject/dao/impl/MarkDaoImpl.java [hunk 7]
```diff
@@ -0,0 +1,208 @@
...
+
+        try (Connection conn = JdbcConnection.getInstance().getConnection();
+             PreparedStatement statement =
+                     conn.prepareStatement(
+                             sql,
+                             Statement.RETURN_GENERATED_KEYS)) {
+
+            statement.setInt(1, nonNullMark.getUserProfileId());
+            statement.setFloat(2, nonNullMark.getScore());
+            statement.setDate(3, new java.sql.Date(nonNullMark.getDate().getTime()));   //check
```
Comment (yanaozembl) [inline src/main/java/com/tproject/dao/impl/MarkDaoImpl.java:132]: change to statement.setDate(3, nonNullMark.getDate);

**Review by yanaozembl — COMMENTED** [review]
Maube it's better to use variable with the name "mark", why did you choose to use "nonNullMark" name?

File: src/main/java/com/tproject/dao/impl/MarkDaoImpl.java (hunk 3 above)
Comment (AntonKolbaska) [reply src/main/java/com/tproject/dao/impl/MarkDaoImpl.java:37]: no, only getDate().

**Pull request merged**